packages/core/src/state-management/anchor-resuming-service.ts

```diff
@@ -0,0 +1,64 @@
+import { StreamID } from '@ceramicnetwork/streamid'
+import { AnchorRequestStoreListResult } from '../store/anchor-request-store.js'
+import { AnchorStatus, DiagnosticsLogger, LogStyle, SyncOptions } from '@ceramicnetwork/common'
+import { Repository } from './repository.js'
+import PQueue from 'p-queue'
+
+const RESUME_QUEUE_CONCURRENCY = 30
+const RESUME_BATCH_SIZE = RESUME_QUEUE_CONCURRENCY * 5
+
+export class AnchorResumingService {
+  /**
+   * Resume running states from anchor request store
+   */
+  readonly resumeQ: PQueue
+
+  /**
+   * true iff the repository is in the process of closing
+   *
+   * @private
+   */
+  #shouldBeClosed = false
+
+  constructor(private readonly logger: DiagnosticsLogger) {
+    this.resumeQ = new PQueue({ concurrency: RESUME_QUEUE_CONCURRENCY })
+  }
+
+  async resumeRunningStatesFromAnchorRequestStore(repository: Repository): Promise<void> {
+    this.#shouldBeClosed = false
+    const currentTimestamp = Date.now()
+    let gt: StreamID | undefined = undefined
+    let batch = new Array<AnchorRequestStoreListResult>()
+    do {
+      batch = await repository.anchorRequestStore.list(RESUME_BATCH_SIZE, gt)
+      batch.map((listResult) => {
+        this.resumeQ.add(async () => {
+          const stream$ = await repository.load(listResult.key, {
+            sync: SyncOptions.PREFER_CACHE,
+          })
+          if ([AnchorStatus.FAILED, AnchorStatus.ANCHORED].includes(stream$.value.anchorStatus)) {
+            await repository.anchorRequestStore.remove(stream$.id)
+            return
+          }
+          // TODO: CDB-2010 Do we want to do the actual anchor request here? Or do we just want to start polling? If the latter, than .load(..) doesn't seem to start polling as of now.
+          await repository.stateManager.anchor(stream$)
+          if (listResult.value.timestamp > currentTimestamp) {
+            throw Error(
+              `System clock possibly moving backwards!!! Check your system clock and if it's accurate, delete your state store`
+            )
+          }
+          this.logger.log(
+            LogStyle.info,
```

Comment:
Suggest using the verbose log level, as this could get quite noisy at startup otherwise if you have a large number of pending anchor requests.